sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StoragePagedFlux.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation;
+
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.PagedResponseBase;
+import com.azure.core.util.paging.PageRetriever;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * This type is a thin wrapper around a PagedFlux that ensures the page size will be requested when they customer passes
+ * the value to byPage method when iterating rather than setting the page size on the options.
+ *
+ * @param <T> The type of items in a {@link PagedResponse}
+ */
+public class StoragePagedFlux<T> extends PagedFlux<T> {
+
+    public StoragePagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever) {
+        super(firstPageRetriever);
+    }
+
+    public static <T> PagedFlux<T> create(Function<Integer, Mono<PagedResponse<T>>> firstPageRetriever,
+        BiFunction<String, Integer, Mono<PagedResponse<T>>> nextPageRetriever) {
+        return PagedFlux.create(() -> (continuationToken, pageSize) ->
+            continuationToken == null
+                ? firstPageRetriever.apply(pageSize).flux()
+                : nextPageRetriever.apply(continuationToken, pageSize).flux());
+    }
```

Comment:
shouldn't this go to core ? @alzimmermsft

Reply:
This would be a good constructor to add into PagedFlux, right now it has the following:

```java
public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever,
        Function<String, Mono<PagedResponse<T>>> nextPageRetriever) {
        this(() -> (continuationToken, pageSize) -> continuationToken == null
            ? firstPageRetriever.get().flux()
            : nextPageRetriever.apply(continuationToken).flux(), true);
    }
```

Reply:
@alzimmermsft If you'd like me to add it to core directly, let me know. And point more towards any samples or tests I'd need to update

Reply:
Given that this is in implementation which is only share between Storage libraries, to unblock the issue I think this can stay and then once Core adds the new APIs it can be transitioned over.

Reply:
I'd rather add this to core directly and depend on unreleased core until next release cycle. IMO it doesn't make sense to create this type and remove it iteration later (think classnotfoundexceptions if someone doesn't get dependnecy versions right...). Unless this issue need some hotfix release (but I think we're not there).

Reply:
This should cover the requirements here: https://github.com/Azure/azure-sdk-for-java/pull/20164